pkg/server/file.go

```diff
@@ -114,24 +114,47 @@ func (t *TranslateServiceServer) UploadTranslationFile(
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
-	switch err := t.repo.SaveMessages(ctx, params.serviceID, messages); {
-	default:
-		// noop
-	case errors.Is(err, repo.ErrNotFound):
-		return nil, status.Errorf(codes.NotFound, "service not found")
-	case err != nil:
-		return nil, status.Errorf(codes.Internal, "")
-	}
+	all := model.MessagesSlice{*messages}
 
-	// If the uploaded file is original and populateTranslations flag is true, populate the translated messages.
-	if messages.Original && params.populateTranslations {
-		allMessages, err := t.repo.LoadMessages(ctx, params.serviceID, repo.LoadMessagesOpts{})
+	// When updating original messages, changes might affect translations - transform and update all translations.
+	if messages.Original {
+		all, err = t.repo.LoadMessages(ctx, params.serviceID, repo.LoadMessagesOpts{})
 		if err != nil {
 			return nil, status.Errorf(codes.Internal, "")
 		}
 
-		if err = t.populateTranslatedMessages(ctx, params.serviceID, messages, allMessages); err != nil {
-			return nil, err
+		originalMessages := messages
+
+		// Replace messages from repo with the new ones, or append if not found.
+		switch idx := all.LanguageIndex(messages.Language); idx {
```

Comment:
why not `.Replace()`?

Reply:
Wanted to delete that method.

The method's behavior can be confusing because it combines two operations: replacement and appending.